packages/app-desktop/gui/getWindowTitle.ts

```diff
@@ -0,0 +1,26 @@
+const Setting = require('@joplin/lib/models/Setting').default;
+import BaseModel from '@joplin/lib/BaseModel';
+import { _ } from '@joplin/lib/locale';
+const { substrWithEllipsis } = require('@joplin/lib/string-utils');
+import { AppState } from '../app';
+
+// notes: any, selectedNoteIds: any, selectedFolderId: any, folders: any, screens: any, route: any
+export default function getWindowTitle(state: AppState) {
```

Comment:
Sorry but it's still not quite right. I wrote this before:

> please use proper types. You can find them in interface AppState

and also this before:

> We need to define what arguments this function takes, so please either create an interface for props that tells what props are actually used by the function, or create separate arguments for each dependency.

But unfortunately you've changed it again and you're now passing the whole application state to the function, making it impossible to know what arguments the function actually needs. As a result, in the tests you are passing "any" for all the variables, so we have no type checks whatsoever.

It would help if you could follow my suggestions or if you think what I'm saying makes no sense, let's discuss it, but there's no point changing the code over and over, each new version being as incorrect as the one before.

Reply:
Hi @laurent22 ,
Sorry for this, being new to ts (and interfaces), I'm mostly guessing how exactly you want the props interface to work. Perhaps it would be great if you could put up some examples on how I should define this interface. Or how I can derive the data from the newProps to the new interface.

Reply:
Hi @asrient, the idea is to narrow down the arguments in the function signature to only what you need, as that makes it easier to test and to know what arguments the function expects. It also generally makes the function more reusable. Now you are passing the whole application state, which means the function can't be used in a context where the application state is not available.

The way you had it in the previous version was fine actually, but you need to replace the "any" type with the real type - i.e. "string", "number", etc. This is so all arguments are type checked. For example `selectedNoteIds: string[]`, `folders: FolderEntity[]`, etc. If you have some doubt about what type should be used let me know, but for the most part you can find the right types in the AppState interface.

Reply:
Upon investigation I found the following types:
```
notes: any[], selectedNoteIds: string[], selectedFolderId: string, folders: any[], screens: any, route: any
```
Please do let me know if any type here is wrong